builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -275,9 +277,14 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 			Key:    aws.String(key),
 		})
 	if err != nil {
-		return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
+		// Treat the inability to get object tags in a restricted regions as a
+		// soft error.
+		if !restricted {
+			return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
```

Comment:
:thinking: Should we also test in `resourceAwsS3BucketObjectPut` if `tags` is set and we're in govcloud, and generate an error there? This way we wouldn't just silently ignore the user's provided tags, but instead tell them very loudly that tags are not supported in GovCloud.